### PR TITLE
Caiso rtpd

### PR DESCRIPTION
**Pull request by ndavis6**
This is the commit Raul and I paired on this morning

### CODE REVIEW

File: tests/test_lmp.py [hunk 1]
```diff
@@ -79,23 +80,43 @@ def test_latest(self):
             self.assertEqual(dp['market'], self.MARKET_CHOICES.fivemin)
             self.assertEqual(dp['freq'], self.FREQUENCY_CHOICES.fivemin)
 
-    def test_date_range(self):
+    def date_range(self, market):
```
Comment (ndavis6) [inline tests/test_lmp.py]: Naming conventions?  Should I add an underscore prefix?

File: pyiso/caiso.py [hunk 2]
```diff
@@ -202,14 +205,17 @@ def get_lmp(self, node_id=None, **kwargs):
...
         for i, row in df.iterrows():
             dp = {
                 'timestamp': i.to_pydatetime(),  # INTERVALSTARTTIME_GMT is the index
-                'lmp': row['LMP_PRC'],
+                'lmp': row['lmp'],
                 'node_id': row['NODE'],
                 'ba_name': 'CAISO',
                 'lmp_type': row['LMP_TYPE'],
-                'market': self.options['market'],
+                'market': invert_oasis_markets[row['MARKET_RUN_ID']],
```
Comment (ndavis6) [inline pyiso/caiso.py]: This could be reverted, I was having problems with market_run_id not reflecting market

File: pyiso/caiso.py [hunk 3]
```diff
@@ -394,6 +400,8 @@ def construct_oasis_payload(self, queryname, **kwargs):
         except KeyError:
             market_run_id = self.oasis_markets[self.options['market']]
 
+        self.options.update(market_run_id=market_run_id)
```
Comment (ndavis6) [inline pyiso/caiso.py]: This fixes the need to manually add market_run_id to certain oasis calls

File: tests/test_lmp.py (hunk 1 above)
Comment (aschn) [inline tests/test_lmp.py]: We don't have a firm convention for this, but I usually do something more like `run_xx_test`

File: pyiso/caiso.py (hunk 2 above)
Comment (aschn) [inline pyiso/caiso.py]: is there a test for the case you're referring to?

**Comment by ndavis6** [comment]
Closes [#42]

File: pyiso/caiso.py (hunk 2 above)
Comment (ndavis6) [inline pyiso/caiso.py]: No, there should be.  But calling get_lmp with market='RTPD' failed and the fix was to add the line below (289), or to add market_run_id='RTPD' to the arguments

**Pull request merged**